### PR TITLE
Fixed Namespace Validation.

### PR DESCRIPTION
**Pull request by ipanova**
[noissue]

### CODE REVIEW

**Comment by ipanova** [comment]
@mdellweg wiht current implementation this error is never raised https://github.com/pulp/pulp_container/pull/680/files#diff-1689ddc49f7d0e21a80ae6006966cfb47766de4a004c7bb1b8e7f442b285384eL342
I think the intention was to allow the user to set the namespace as long as it matches the first portion of the base_path

**Comment by ipanova** [comment]
I've also noticed we have 0 tests for namespaces so I ought to add some as part of this PR

**Review by mdellweg — APPROVED** [review]

**Comment by ipanova** [comment]
https://github.com/kit-oz/drf-complex-filter
https://stackoverflow.com/questions/16974862/complex-filters-in-django-model

**Review by mdellweg — APPROVED** [review]

File: pulp_container/app/serializers.py [hunk 1]
```diff
@@ -300,6 +300,7 @@ class ContainerDistributionSerializer(DistributionSerializer):
     namespace = RelatedField(
         required=False,
         read_only=True,
+        allow_null=False,
```
[the remark this answers is not in the document]
Comment (ipanova) [reply pulp_container/app/serializers.py]: allow_null defaults to False, this is redundant in any case.

**Pull request merged**